### PR TITLE
lite2: rename alternative providers to witnesses

### LINKED ISSUE #4341: lite2: rename Alternatives to Witnesses
The reason is alternative provider may be viewed as something not important. But in reality we are using alternative providers for cross-checking headers and they are super important.

Other solution may be requiring at least 5 (don't know what the good number is) alternative providers.

### PR DESCRIPTION
**Pull request by melekes**
Closes #4341

* [x] Referenced an issue explaining the need for the change
* [ ] ~~Updated all relevant documentation in docs~~
* [x] Updated all code comments where relevant
* [x] Wrote tests
* [ ] ~~Updated CHANGELOG_PENDING.md~~

### CODE REVIEW

File: lite2/client.go [hunk 1]
```diff
@@ -84,11 +84,12 @@ func SkippingVerification(trustLevel tmmath.Fraction) Option {
 	}
 }
 
-// AlternativeSources option can be used to supply alternative providers, which
-// will be used for cross-checking the primary provider.
-func AlternativeSources(providers []provider.Provider) Option {
+// Witnesses option can be used to supply providers, which will be used for
+// cross-checking the primary provider. A witness can become a primary iff the
```
Comment (cmwaters) [inline lite2/client.go]: *picky:* A witness can become a primary iff -> A witness can become a primary if

File: lite2/client.go [hunk 2]
```diff
@@ -84,11 +84,12 @@ func SkippingVerification(trustLevel tmmath.Fraction) Option {
...
+// current primary is unavailable.
+func Witnesses(providers []provider.Provider) Option {
 	return func(c *Client) {
-		c.alternatives = providers
```
Comment (cmwaters) [inline lite2/client.go]: Just as a thought. Do we want to check if these are on the same chain - i.e. `c.chainID == provider.ChainID()`. Also could be a way of initially checking if the provider is responding. Could be unnecessary thought.

File: lite2/client.go (hunk 1 above)
Comment (melekes) [reply lite2/client.go]: iff - if and only if

Comment (cmwaters) [reply lite2/client.go]: ah I see. Thanks :)

File: lite2/client.go (hunk 2 above)
Comment (melekes) [reply lite2/client.go]: Good idea! We should do this upon startup. Want to do it? https://github.com/tendermint/tendermint/issues/4350

**Pull request merged**